### PR TITLE
Inline inputs save on blur

### LINKED ISSUE #11567: Save renamings when clicking out of text fields
### Context
When renaming maps, layers or widgets ENTER should be typed in order to save the changes. We've observed a big percentage of users expect those changes to be autosaved by clicking out of the text field.
 
### Steps to Reproduce

1. Double click into a map name, layer name or widget name.
2. Type a new name
3. Clic out of the box

### Current Result
Name is not changed
 
### Expected result
Name should be changed to what has been typed

### PR DESCRIPTION
**Pull request by rjimenezda**
Fix #11567 

Inline inputs now save on blur, and restore previous value when hitting escape, without triggering a save.

### CODE REVIEW

**Comment by rubenmoya** [comment]
In this case:
1. User double click on the title
2. User clicks outside of the title without editing

update_title

I think we should close the input, it could be confusing if you click outside the input and it doesn't close because it hasn't changed.

**Comment by rjimenezda** [comment]
Hmmm, that's not intended. I'll give it a look

**Comment by rjimenezda** [comment]
Fixed, hopefully

File: lib/assets/core/javascripts/cartodb3/components/inline-editor/inline-editor-view.js [hunk 1]
```diff
@@ -92,18 +103,32 @@ module.exports = CoreView.extend({
...
+  blur: function () {
+    this.save();
+    this.hide();
+  },
+
+  save: function () {
+    var value = this.getValue();
+
+    if (value !== this._oldText) {
+      this._onEdit && this._onEdit(this.getValue());
```
Comment (rubenmoya) [inline lib/assets/core/javascripts/cartodb3/components/inline-editor/inline-editor-view.js]: We already have `this.getValue()` in `value`, can't we use it here?

**Comment by ivanmalagon** [comment]
Found an issue.

The `inline-editor-view` is also used in dataset page. The result of that input gets sanitized. If we enter some text that results in an empty string, the confirmation dialog shows up, then it tries to rename the dataset and it fails.

Well, I see now that it's not needed to enter HTML to be sanitized. It fails also with an empty input. And it fails in production, so it's not been added in this PR but it'd nice to fix it, taking advantage of the Pisuerga river passing by Valladolid.

script error

**Comment by rubenmoya** [comment]
I've been messing around with the input, and found this:

input_save

- Using `I <3 maps` it removes the everything but the I.
- Using `A > B` changes the symbol with the sanitised value `&gt;`
- Using `<script>alert('asdasd')</script>` doesn't save using enter, but doesn't give any feedback either.

I'm not sure if that's the behaviour we want, or if we want to improve it somehow. What do you think?

**Comment by ivanmalagon** [comment]
That's all due to sanitization steps.

- `I <3 maps` gets cut from the the tag opening.
- The `script` example gets totally wiped out and since an empty string is not a valid input, the control does nothing.
- The `&gt;` perhaps can be improved, yes.

Anyway, IMHO I won't spend time on it since I think they're all corner cases. Opening / closing tags are not a common option in naming maps and it's not straightforward to improve the feedback about it. I don't think it's worth the effort.

**Comment by rubenmoya** [comment]
Another thing I found, if you press enter in the confirmation dialog, it shows up again, it's not happening in production:

multiple_enter

**Comment by rjimenezda** [comment]
Oh, that's bad. Put it back in development please.

**Comment by rubenmoya** [comment]
Nevermind, it's happening in production too, but only in Safari.

**Comment by rubenmoya** [comment]
I've checked it and `.save()` it's only called once

**Comment by rjimenezda** [comment]
Ohhh we still have the debounce, that's why. Ok 🆒

**Pull request merged**